tests/test_notifications.py

```diff
@@ -107,3 +109,24 @@ def test_translated_from_email(notification_template, settings, language):
     assert message.from_email == settings.ILMOITIN_TRANSLATED_FROM_EMAIL.get(
         language, settings.DEFAULT_FROM_EMAIL
     )
+
+
+def test_notification_delayed_sending(notification_template):
+    context = {
+        "extra_var": "foo",
+        "subject_var": "bar",
+        "body_html_var": "html_baz",
+        "body_text_var": "text_baz",
+    }
+    # Override settings to use django-mailer email backend
+    settings.EMAIL_BACKEND = "mailer.backend.DbBackend"
+    settings.MAILER_EMAIL_BACKEND = "django.core.mail.backends.locmem.EmailBackend"
+
+    send_notification("foo@bar.fi", "event_created", context, "fi", delayed=True)
+    assert len(mail.outbox) == 0
+    assert Message.objects.count() == 1
+    # Now actually send emails
```

Comment:
how about tryng to use just `send_notification("foo@bar.fi", "event_created", context, "fi")`, this way you will check that it uses the if clause, when needed. you will have now 2 emails, but that's probably ok for the testing purposes 😅

Reply:
Actually there is already a separated test for `send_notification` without `delayed=True` above :D so I think it's not really necessary here

Reply:
but it does not use the same `EMAIL_BACKEND`, so no `Message` object from mailer is created there.

Reply:
emails there just go straight to `mail.outbox`

Reply:
Aha, makes sense 😃I'll update it

Reply:
for the record I had in mind this:
```python
send_notification("foo@bar.fi", "event_created", context, "fi", delayed=True)
assert len(mail.outbox) == 0
assert Message.objects.count() == 1
# Now actually fire the one that sends emails
send_notification("foo@bar.fi", "event_created", context, "fi")
assert Message.objects.count() == 0
assert len(mail.outbox) == 2
```
but you solution checks for the same things essentially 👍 you can keep whichever one you like. I am approving this now 🙂

Reply:
Now after reading your solution. I just realized that if there are some notifications in the queue, any `send_notification` without `delayed=True` will flush the queue and send massive emails. Meaning you have to stick with `delayed` in every notification. Otherwise, the queue will behave in a weird way

Reply:
yes, that is true. devs should either keep it mind, OR there could be a totally different function `queue_notification`

Reply:
That is a very good point. I think ideally it should be possible to use both instant and queued notifications in the same project, but based on very brief investigation of django-mailer this might not be easily done.

If we limiting the usage for one type only, I like Igor's idea of having a separate function to underline the separation and hopefully decrease change of usage errors. Another option could be to use the exact same function, but make the choice by a global setting. That way it would not be possible to use more than one type by mistake. Not sure if I actually like that way or not though :)

Reply:
I don't think we can use both `send` and `queue` notification in the same project with the current implementation of `send_notification` and I'm not sure if there is anyway to do that with django-mailer. So I prefer the global setting solution as it guarantees that there won't be any usage mistake. Will update the PR